playground/miden-wasm/src/lib.rs

```diff
@@ -63,16 +63,62 @@ pub fn program(asm: &str, inputs_frontend: &str, output_count: u16) -> Vec<u64>
     )
     .map_err(|err| format!("Failed to generate exection trace = {:?}, and advice_tape = {:?}", err, advice_tape)).unwrap();
     
-    // TODO: Investigate why proof verification fails when outputCount > 1
-    //assert!(miden::verify(*program.hash(), &[], &outputs, proof).is_ok());
-    // tracked in https://github.com/0xPolygonMiden/examples/issues/19
     trace.program_outputs().stack_outputs(output_count as usize).to_vec()
 }
 
-/// Basic test for the program
+/// Proves the program with the given inputs
+#[wasm_bindgen]
+pub fn prove_program(asm: &str, inputs_frontend: &str, output_count: u16) -> Vec<u64> {
+    console_error_panic_hook::set_once();
+    wasm_logger::init(wasm_logger::Config::default());
+
+    let assembler = Assembler::new();
+    let program = assembler.compile(&asm).expect("Could not compile source");
+    
+    let mut stack_init = <Vec<u64>>::new();
+    let mut advice_tape = <Vec<u64>>::new();
+
+    // TODO: We must catch all cases of inputs. Input can be not a valid json,
+    // input can be empty, stack_init can be empty, advice_tape can be empty. And all combinations.
```

Comment:
nit: I see - you're talking about cases of input errors. I would probably have said "all cases of input errors", but the message is now clear, and no need to update, especially since it'll be tackled in the next PR.

Reply:
ah indeed. I like it those nits